builtin/providers/github/util_collaborators.go

```
@@ -0,0 +1,112 @@
+package github
+
+import "github.com/google/go-github/github"
+
+// removeInvitation removes invitation for user if it exists.
+func removeInvitation(client *github.Client, owner, repository, invitee string) error {
+	repoID, err := repositoryID(client, owner, repository)
+	if err != nil {
+		return err
+	}
+
+	i, err := getRepositoryInvitation(client, *repoID, owner, invitee)
+	if err != nil {
+		return err
+	}
+	if i == nil {
+		return nil
+	}
+
+	_, err = client.Repositories.DeleteInvitation(*repoID, *i.ID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// permissionsForCollaboratorOrInivitee returns user's permission if he is collaborator, invitation's
+// permission if he is invitee or empty string otherwise.
+func permissionsForCollaboratorOrInivitee(client *github.Client, owner, repo, invitee string) (string, error) {
+	p, err := permissionsForCollaborator(client, owner, repo, invitee)
+	if err == nil && p != "" {
```

if there an error, it would be silently ignored. IMO the pattern should be always

```go
p, err := permissionsForX(client, owner, repo, username)
if err != nil {
  return "", err
}
if p != "" {
  return p, nil
}
```